class.jetpack-gutenberg.php

```diff
@@ -188,6 +188,20 @@ public static function set_extension_available( $slug ) {
 	 * @param array  $details A free-form array containing more information on why the extension is unavailable.
 	 */
 	public static function set_extension_unavailable( $slug, $reason, $details = array() ) {
+		/**
+		 * Filter 'jetpack_block_editor_enable_upgrade_nudge' with `true` to enable or `false`
+		 * to disable paid feature upgrade nudges in the block editor.
+		 *
+		 * @since 7.7.0
+		 *
+		 * @param boolean
+		 */
+		if ( 'missing_plan' === $reason && ! apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ) ) {
```

Comment:
I like this, but I would move the docblock directly with the filter, by splitting the condition onto 2 different lines. I think it may be a bit more readable?

```php
if (
	// Here we could even have a comment explaining this: https://github.com/Automattic/jetpack/pull/13228/files#r314138211
	'missing_plan' === $reason
	/**
	 * Filter 'jetpack_block_editor_enable_upgrade_nudge' with `true` to enable or `false`
	 * to disable paid feature upgrade nudges in the block editor.
	 *
	 * @since 7.7.0
	 *
	 * @param boolean
	 */
	&& ! apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false )
) {
```

What do you think?

Reply:
👍 Good idea. Applied in 55b075a7976b7d0b5c5f955a284a0c369cf2082e